suite-native/qr-code/src/components/PickQRFromGalleryButton.tsx

```diff
@@ -0,0 +1,43 @@
+import { BarcodeScanningResult, scanFromURLAsync } from 'expo-camera';
+import * as ImagePicker from 'expo-image-picker';
+
+import { Button } from '@suite-native/atoms';
+import { Icon } from '@suite-common/icons-deprecated';
+import { useToast } from '@suite-native/toasts';
+import { Translation } from '@suite-native/intl';
+
+type PickQRFromGalleryButtonProps = {
+    onImagePicked: (data: BarcodeScanningResult) => void;
+};
+
+export const PickQRFromGalleryButton = ({ onImagePicked }: PickQRFromGalleryButtonProps) => {
+    const { showToast } = useToast();
+
+    const handlePickImage = async () => {
+        const pickedImage = await ImagePicker.launchImageLibraryAsync({});
+        const imageUri = pickedImage?.assets?.[0].uri;
+
+        try {
+            const scannedResults = await scanFromURLAsync(imageUri!, ['qr']);
+            const QrData = scannedResults[0];
+
+            onImagePicked(QrData);
+        } catch (error) {
+            showToast({
```

Comment:
Oh, you are right. It is a known issue, but it definitely does not look nice here. I changed it so the bottom sheet is hidden in case of error, so the toast is visible.